python-ci/pipeline.yaml

```diff
@@ -0,0 +1,37 @@
+env:
+  PYTHON_VERSION: "3"
+
+steps:
+  - label: ":pip: Install dependencies"
+    key: "pip"
+    command: pip install -r requirements.txt
+    plugins:
+      - docker#v5.9.0:
+          image: "python:${PYTHON_VERSION}"
+
+  - label: Lint with Ruff
+    depends_on: ["pip"]
+    command: |
+      pip install ruff
+      ruff check .
+    plugins:
+      - docker#v5.9.0:
+          image: "python:${PYTHON_VERSION}"
+
+  - label: ":pytest: Run pytest"
+    key: "pytest"
+    depends_on: ["pip"]
+    command: |
+      pip install pytest
+      pytest --junitxml=junit/test-results.xml \
+    artifact_paths:
```

Comment:
Should we elaborate on artifacts on the README and link to the Docs page?

Reply:
Perhaps 🤷 The usage here was more out a necessity for using the junit annotate plugin which expects artifacts to be available for download.